classes/Dispatcher/EventDispatcher.php

```diff
@@ -77,7 +77,8 @@ public function __construct(
     public function dispatch($name, array $params)
     {
         // Events are related to actions on the shop, not the back office
-        if (!in_array($this->context->controller->controller_type, ['front', 'modulefront'])) {
+        $controller = $this->context->controller;
```

Comment:
```suggestion
        /** @var \Controller $controller */
        $controller = $this->context->controller;
```

This may solve the issue.